pycontrails/core/met.py

```diff
@@ -2625,25 +2627,47 @@ def _add_vertical_coords(data: XArrayType) -> XArrayType:
     # It is more important for air_pressure and altitude to be grid-aligned than to be
     # coordinate-aligned, so we use the dtype of the data to determine the precision of
     # these coordinates
-    if isinstance(data, xr.Dataset):
-        dtype = np.result_type(*data.data_vars.values(), np.float32)
+    dtype = (
+        np.result_type(*data.data_vars.values(), np.float32)
+        if isinstance(data, xr.Dataset)
+        else data.dtype
+    )
+
+    coords = data.coords
+    try:
+        air_pressure = coords["air_pressure"]
+        altitude = coords["altitude"]
+    except KeyError:
+        pass
     else:
-        dtype = data.dtype
+        if air_pressure.dtype == dtype and altitude.dtype == dtype:
+            return data
+        # At least one of 'air_pressure' or 'altitude' has the wrong dtype
+        data.coords["air_pressure"] = air_pressure.astype(dtype, copy=False)
+        data.coords["altitude"] = altitude.astype(dtype, copy=False)
+        return data
+
```

Comment:
Is this try-else block really necessary?

I think you could simplify the code by deleting it and changing the if-else blocks below it to
```python
if "air_pressure" not in coords:
   ...
elif data.coords["air_pressure"].dtype != dtype:
   ...
```
Maybe there is a corner case I haven't thought of, though.

Reply:
I agree - there may be a slight performance improvement with try-except, but for clarity, I think we should handle key addition and dtype all in the same place

Reply:
https://github.com/contrailcirrus/pycontrails/pull/216/commits/e48adb2a37b3cfa06d551f8546a9c39493f342c9